admin/broadleaf-open-admin-platform/src/main/java/org/broadleafcommerce/openadmin/web/service/FormBuilderServiceImpl.java

```diff
@@ -514,7 +514,7 @@ public ListGrid buildCollectionListGrid(String containingEntityId, DynamicResult
             } else {
                 for (String fieldName : atcmd.getGridVisibleFields()) {
                     Property p = cmd.getPMap().get(fieldName);
-                    if (p != null) {
+                    if (propertyExistsInResultSet(p, drs)) {
```

Comment:
if property doesn't exist on entity I would think that it is not in class metadata and so cmd.getPMap.get(fieldName) should return null?
If we query less fields than in metadata this fix should address it, but I don't think we do this OOB ?

Reply:
I looked for the possibility of using the class metadata.  Unfortunately, the class metadata "inherits" some properties from the adornedTargetCollection and in this case it has the adorned fields.